app/kumactl/cmd/get/get_mesh.go

```diff
@@ -27,13 +28,12 @@ func newGetMeshCmd(pctx *getContext) *cobra.Command {
 				return err
 			}
 			name := args[0]
-			currentMesh := name
 			mesh := &core_mesh.MeshResource{}
-			if err := rs.Get(context.Background(), mesh, store.GetByKey(name, currentMesh)); err != nil {
+			if err := rs.Get(context.Background(), mesh, store.GetByKey(name, model.NoMesh)); err != nil {
```

Comment:
I tried this at first but to be honest I don't like this if you look closely at the API of the store.

I think `CreateBy...` should have a full list of arguments that are the key to the resource. Mesh is a part of the key, it's just empty. So for create, I'd rather stay with `CreateByKey`

For `Get`, if we introduce `GetByName` then it will be possible to call `GetByName` on the resource which is bound to a Mesh which I don't think is right. Additionally, there will be an asymmetry between `CreateBy` and `GetBy`.

Also when we add a new field as a part of the Key, `GetByKey` and `CreateByKey` will force us to go through all the usages

Reply:
Yeah okay, let's stick to this